File: 0016-3sum-closest/0016-3sum-closest.py
class Solution:
    def threeSumClosest(self, nums: List[int], target: int) -> int:
        nums.sort()
        clo_sum=sum(nums[:3])

        for i in range(len(nums)):
            if i>0 and nums[i]==nums[i-1]: continue
            j=i+1
            k=len(nums)-1
            while j<k:
                total=nums[i]+nums[j]+nums[k]
                if total==target: return total
                elif abs(total-target)<abs(clo_sum-target): clo_sum=total
                if total<target: j+=1
                else: k-=1
        return clo_sum

File: 0016-3sum-closest/0016-3sum-closest.c
#include<stdlib.h>
#include<math.h>

int compare(const void* a, const void* b) {
    return (*(int*)a - *(int*)b);
}

int threeSumClosest(int* nums, int numsSize, int target) {
    qsort(nums,numsSize,sizeof(int),compare);
    int clo_sum=nums[0]+nums[1]+nums[2];

    for(int i=0;i<numsSize-2;i++){
        int j=i+1;
        int k=numsSize-1;

        while(j<k){
            int total=nums[i]+nums[j]+nums[k];

            if (total==target)
                return total;
            
            if (abs(total-target)<abs(clo_sum-target))
                clo_sum=total;
            if (total<target)
                j++;
            else
                k--;
        }
    }
    return clo_sum;
}